source/patterns/@aws-solutions-constructs/aws-fargate-stepfunctions/test/fargate-stepfunctions.test.ts

```diff
@@ -0,0 +1,463 @@
+    vpcProps: { cidr: testCidr },
+    clusterProps: { clusterName },
+    containerDefinitionProps: { containerName },
+    fargateTaskDefinitionProps: { family: familyName },
+    fargateServiceProps: { serviceName },
+    stateMachineProps: testStateMachineProps(stack),
+    stateMachineEnvironmentVariableName: customEnvName
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::TaskDefinition", {
+    Family: familyName,
+    ContainerDefinitions: [
+      {
+        Environment: [
+          {
+            Name: customEnvName,
+            Value: {
+              Ref: "testconstructStateMachine3333AAA9"
+            }
+          }
+        ],
+        Essential: true,
+        Image: {
+          "Fn::Join": [
+            "",
+            [
+              "123456789012.dkr.ecr.us-east-1.",
+              {
+                Ref: "AWS::URLSuffix"
+              },
+              "/fake-repo:latest"
+            ]
+          ]
+        },
+        MemoryReservation: 512,
+        Name: containerName,
+        PortMappings: [
+          {
+            ContainerPort: 8080,
+            Protocol: "tcp"
+          }
+        ]
+      }
+    ]
+  });
+});
+
+test('Check for no cloudwatch creation', () => {
+  const stack = new cdk.Stack();
+  const publicApi = true;
+
+  const construct = new FargateToStepfunctions(stack, 'test-construct', {
+    publicApi,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+    vpcProps: { cidr: testCidr },
+    clusterProps: { clusterName },
+    containerDefinitionProps: { containerName },
+    fargateTaskDefinitionProps: { family: familyName },
+    fargateServiceProps: { serviceName },
+    stateMachineProps: testStateMachineProps(stack),
+    createCloudWatchAlarms: false
+  });
+
+  expect(construct.cloudwatchAlarms).not.toBeDefined();
+  expect(stack).not.toHaveResource("AWS::CloudWatch::Alarm", cloudwatchTest());
+});
+
+test('Check for custom log group props', () => {
+  const stack = new cdk.Stack();
+  const publicApi = true;
+  const logGroupName = "custom-log-group";
+
+  new FargateToStepfunctions(stack, 'test-construct', {
+    publicApi,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+    vpcProps: { cidr: testCidr },
+    clusterProps: { clusterName },
+    containerDefinitionProps: { containerName },
+    fargateTaskDefinitionProps: { family: familyName },
+    fargateServiceProps: { serviceName },
+    stateMachineProps: testStateMachineProps(stack),
+    logGroupProps: {
+      logGroupName
+    }
+  });
+
+  checkForBaseResources(stack);
+  expect(stack).toHaveResourceLike("AWS::Logs::LogGroup", {
+    LogGroupName: logGroupName
+  });
+});
+
+function testStateMachineProps(stack: cdk.Stack, userProps?: stepfunctions.StateMachineProps):
+  stepfunctions.StateMachineProps {
+  const defaultTestProp = { definition: new stepfunctions.Pass(stack, 'StartState') };
+
+  return defaults.consolidateProps(defaultTestProp, userProps);
+}
+
+function serviceTest() {
```

Comment:
Saw all these after i saw CheckForBaseResources() below. Same principle applies - if we don't want to replicate this extensive code/json, then let's rearrange the tests so it only appears once. For these especially, at some point a change in a single test will cause that test to start to fail and we'll have to unwind this change for that test.